### PR TITLE
refactor: use blob share version for padding shares

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/2142

### CODE REVIEW

File: pkg/shares/split_sparse_shares_test.go [hunk 1]
```diff
@@ -38,3 +39,38 @@ func TestSparseShareSplitter(t *testing.T) {
 	got := sss.Export()
 	assert.Len(t, got, 2)
 }
+
+func TestWriteNamespacePaddingShares(t *testing.T) {
+	ns1 := appns.MustNewV0(bytes.Repeat([]byte{1}, appns.NamespaceVersionZeroIDSize))
+	blob1 := newBlob(ns1, appconsts.ShareVersionZero)
```
Comment (evan-forbes) [inline pkg/shares/split_sparse_shares_test.go:45]: could we use a different more arbitrary share version here without breaking anything? The reasoning being that the test as is would have passed before this change, but different share versions wouldn't have.

**Review by evan-forbes — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]

File: pkg/shares/split_sparse_shares_test.go (hunk 1 above)
Comment (rootulp) [reply pkg/shares/split_sparse_shares_test.go:45]: I attempted to but it fails for a different reason earlier in the test case b/c every other share version is unsupported.

Comment (rootulp) [reply pkg/shares/split_sparse_shares_test.go:45]: TBH I'm unsatisfied too b/c I want a test case with a different share version.

Comment (rootulp) [reply pkg/shares/split_sparse_shares_test.go:45]:
```
	t.Run("with a different share version", func(t *testing.T) {
		ns2 := appns.MustNewV0(bytes.Repeat([]byte{2}, appns.NamespaceVersionZeroIDSize))
		blob1 := newBlob(ns2, 1)

		sss := NewSparseShareSplitter()

		err := sss.Write(blob1)
		assert.NoError(t, err)
	})
```

```
--- FAIL: TestWriteNamespacePaddingShares (0.00s)
    --- FAIL: TestWriteNamespacePaddingShares/with_a_different_share_version (0.00s)
        /Users/rootulp/git/rootulp/celestia/celestia-app/pkg/shares/split_sparse_shares_test.go:75: 
            	Error Trace:	/Users/rootulp/git/rootulp/celestia/celestia-app/pkg/shares/split_sparse_shares_test.go:75
            	Error:      	Received unexpected error:
            	            	unsupported share version: 1
            	Test:       	TestWriteNamespacePaddingShares/with_a_different_share_version
        /Users/rootulp/git/rootulp/celestia/celestia-app/pkg/shares/split_sparse_shares_test.go:77: 
            	Error Trace:	/Users/rootulp/git/rootulp/celestia/celestia-app/pkg/shares/split_sparse_shares_test.go:77
            	Error:      	Received unexpected error:
            	            	cannot write namespace padding shares on an empty SparseShareSplitter
            	Test:       	TestWriteNamespacePaddingShares/with_a_different_share_version
FAIL
FAIL	github.com/celestiaorg/celestia-app/pkg/shares	1.208s
FAIL
```

**Pull request merged**